datafusion/physical-expr/src/scalar_function.rs

```diff
@@ -140,18 +140,23 @@ impl PhysicalExpr for ScalarFunctionExpr {
             .map(|e| e.evaluate(batch))
             .collect::<Result<Vec<_>>>()?;
 
+        let inputs_all_scalars = !inputs.is_empty() && inputs
+            .iter()
+            .all(|arg| matches!(arg, ColumnarValue::Scalar(_)));
+
         // evaluate the function
-        let output = self.fun.invoke_batch(&inputs, batch.num_rows())?;
+        let output = self.fun.invoke_with_args(ScalarFunctionArgs {
+            args: inputs,
```

Comment:
the whole point of this PR is to pass in `inputs` here rather than inputs.as_ref

Reply:
Eventually we should probably consider doing the same thing for `evaluate` 🤔